src/schema/messaging/send.ts

```diff
@@ -0,0 +1,26 @@
+import { Type } from "@sinclair/typebox";
+
+import { defineEndpoint } from "@/generator-helpers.js";
+
```

Comment:
For reconnect, the client should just send message again and assumption is that message copies are not a big deal?

Reply:
not sure I follow, which reconnect? If the client never received the response to `messaging/send` then yes, it should send the message again and that may result in duplicate, but that's not a big deal indeed.

I haven't really put much thought into guaranteeing end-to-end delivery of messages, do you think we need that?

Reply:
Agreeing on fire-and-forget/at-least-once/exactly-once is good idea. I believe at-least-once would be good to have, but if you want we can also start with what you purpose and rework later.

Reply:
I think the easiest way for it to be at-least-once is make it a server->client request, not event, and retry in the server in the "session" process (not the TCP connection one). To allow making it exactly-once I guess you could add a increasing id/time-stamp to the message and let the recipients client deduplicate.

Reply:
(I assume we do not want to have a history or anything like that yet)

Reply:
ok, let's make the `server -> client` part also a request so that the server knows when a message is safe to delete. At least the proto will support at-least-once delivery, that would be a pain to change later.

Reply:
Although, the at least once delivery is going to be problematic for multicast messages, typically for parties and lobbies. I think it will add a ton of complexity there :/